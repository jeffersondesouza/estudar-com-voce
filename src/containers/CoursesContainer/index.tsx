import * as React from 'react';
import { connect } from 'react-redux';

import './style/style.scss';
import FeedMidleware from '../../store/modules/course/middleware';

import { Search } from '../../components/Search';
import { CoursesList } from '../../components/Courses-list';
import { Course } from '../../models/Course';


export interface IProps {
  dispatchloadCourses(): void,
  genericCourses?: Course[]
  customCourses?: Course[]
}

export class CoursesContainer extends React.Component<IProps> {

  componentDidMount() {
    this.props.dispatchloadCourses();
  }

  render() {
    const { genericCourses, customCourses } = this.props;
    
    return (
      <div className="courses-block">
        <Search />
        <CoursesList courses={customCourses || []} title="Materias Personalizadas sem faculdade" />
        <CoursesList courses={genericCourses || []} title="Materias Gerais" />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  ...state.course,
});

const mapDispatchToProps = dispatch => ({
  dispatchloadCourses: () => dispatch(FeedMidleware.loadCoursesRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(CoursesContainer);
