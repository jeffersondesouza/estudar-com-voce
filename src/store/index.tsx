/**
 * Store configuration - Here we configure the application store, making it avalable to the projt by simporting this module 
 */
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from './modules/index';

export const configure = (initialState = {}) => (
    createStore(
        reducer,
        initialState,
        compose(
            applyMiddleware(thunk),
        )
    )
);

export default configure();
