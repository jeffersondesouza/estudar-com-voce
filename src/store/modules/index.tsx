import { combineReducers } from 'redux';

import  courseReducer from './course/reducer';

const rootReducer = combineReducers({
  course:  courseReducer,
});

export default rootReducer;
