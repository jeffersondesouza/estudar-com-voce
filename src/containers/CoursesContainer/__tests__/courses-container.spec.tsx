import * as React from 'react';
import axios from 'axios';

import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';

import * as renderer from 'react-test-renderer';

import store from '../../../store';
import CoursesContainer from './../';
import ResMock from '../../../../__mocks__/courses.mock';

jest.mock('axios');

describe('<CoursesContainer>', () => {

  describe('when component mount', () => {

    it('should call and list courses', () => {

      const resp = { data: ResMock };
      const component = <CoursesContainer dispatchLoadFeed={() => { }} />;

      (axios.get as any).mockResolvedValue(resp);

      const container = (
        <Provider store={store}>
          <StaticRouter context={{}}>{component}</StaticRouter>
        </Provider>
      );
      const tree = renderer.create(container).toJSON();
      expect(tree).toMatchSnapshot();
      expect(axios.get).toHaveBeenCalled();


      
    });

  });


});


/*

HttpService.get = jest.fn();


describe('<CoursesContainer>', () => {

  describe('when CoursesContainer mount', () => {

    context('when server respond feed request with SUCCESS', () => {
      it('should match the component spnapshot', () => {

        HttpService.get.mockResolvedValue(USERS_MOCK);

        const component = (
          <Provider store={store}>
            <StaticRouter context={{}}>
              <CoursesContainer />
            </StaticRouter>
          </Provider>
        );

        const tree = renderer.create(component).toJSON()
        const wrapper = mount(component);

        expect(tree).toMatchSnapshot();
        expect(HttpService.get).toHaveBeenCalled();

      });
    });

    context('when server respond feed request with ERROR', () => {
      it('should match the component spnapshot', () => {

        HttpService.get.mockRejectedValue(new Error('could not load the components'));

        const component = (
          <Provider store={store}>
            <StaticRouter context={{}}>
              <CoursesContainer />
            </StaticRouter>
          </Provider>
        );

        const tree = renderer.create(component).toJSON()
        const wrapper = mount(component);

        expect(tree).toMatchSnapshot();
        expect(HttpService.get).toHaveBeenCalled();

      });
    });

  });

});


 */