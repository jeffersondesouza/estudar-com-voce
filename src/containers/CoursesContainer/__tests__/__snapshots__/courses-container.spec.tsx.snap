// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoursesContainer> when component mount should call and list courses 1`] = `
<div
  className="courses-block"
>
  <div
    className="search"
  >
    <div
      className="search__input"
    >
      <input
        className="form-control"
        placeholder="Informe o nome do curso que deseja encontrar"
      />
    </div>
    <div
      className="search__filters"
    >
      <div
        className="search__radio-label"
      >
        <input
          id="ipt1"
          name="filter"
          type="radio"
        />
        <label
          htmlFor="ipt1"
        >
          Todos
        </label>
      </div>
      <div
        className="search__radio-label"
      >
        <input
          id="ipt2"
          name="filter"
          type="radio"
        />
        <label
          htmlFor="ipt2"
        >
          Geral
        </label>
      </div>
      <div
        className="search__radio-label"
      >
        <input
          id="ipt3"
          name="filter"
          type="radio"
        />
        <label
          htmlFor="ipt3"
        >
          Especifico
        </label>
      </div>
      <i
        className="icon-bank"
      />
    </div>
  </div>
  <div
    className="courses"
  >
    <h2
      className="title courses__title"
    >
      Materias Personalizadas sem faculdade
    </h2>
    <ul
      className="courses__list"
    >
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Física II
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Cálculo IV
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Física IV
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Álgebra Linear II
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Eletricidade Geral
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Cálculo II
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Estatística
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Cálculo III
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Física III
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Física I
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header course__header--generic"
          >
            <h3>
              Álgebra Linear I
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="courses"
  >
    <h2
      className="title courses__title"
    >
      Materias Gerais
    </h2>
    <ul
      className="courses__list"
    >
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header"
          >
            <h3>
              Probabilidade
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header"
          >
            <h3>
              Mecânica I
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            >
              Novo!
            </div>
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header"
          >
            <h3>
              Programação em Python
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
      <li
        className="courses__list-item"
      >
        <div
          className="course"
        >
          <div
            className="course__header"
          >
            <h3>
              Cálculo Numérico
            </h3>
          </div>
          <div
            className="course__footer"
          >
            <div
              className="course__highlight"
            />
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
